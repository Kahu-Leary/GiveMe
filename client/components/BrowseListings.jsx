import React, { useState, useEffect } from 'react'

import { getItems } from '../apiClient'

import Item from './Item'

export default function BrowseListings() {
  const [items, setItems] = useState({
    items: []
  })
  useEffect(() => {
    getItems()
      .then((res) => {
        return setItems(res)
      })
      .catch((error) => {
        console.log('error: ', error.message)
      })
  }, [])
  return (
    <>
      <div>
        <ul className="imageGridContainer" >
          {items.items.map((item) => {
            return <li className="imageGridItem" key={item.id}>
              <Item key={item.id} item={item} /></li>
          })}
        </ul>
      </div>
    </>
  )
}
