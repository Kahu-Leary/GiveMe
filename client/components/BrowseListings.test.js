import React from 'react'

import renderer from 'react-test-renderer'

import BrowseListings from './BrowseListings'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowseListings/>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
