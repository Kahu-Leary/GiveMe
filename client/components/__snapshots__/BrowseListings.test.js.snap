// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="columns"
  >
    <label
      className="label "
    >
      Search
    </label>
    <input
      className="input "
      id="searchbar"
      name="searchbar"
      placeholder="Search"
      type="text"
    />
  </div>
</div>
`;
